### DISCUSSION TITLE
Argocd rollouts return different app versions

### DISCUSSION THREAD

**Discussion by Spartaques** [argoproj/argo-rollouts#2053] [Q&A]
Hello guys. I have a frontend app using React js. And i'm using Argocd Rollouts with NGINX Controller for a/b testing. But for now, it works in random order. index.html returns from one version, but all assetrs returns from another version. So, if there any strategy to using client ip address for example as key for traffic routing or another solutions for this? Maybe i'm doing smth wrong. Helm please

**Comment by kostis-codefresh** [comment]
Hello

You need to look at the documentation of nginx on how to do a proper split. Create manually 2 services and try to make each service redirect to exactly where you want. After you have this working, THEN add Argo Rollouts in the mix.

Argo Rollouts delegates all traffic decisions to the traffic provider. It doesn't do any traffic splits on its own.
